### ISSUE TITLE
how to set ignore_label in tensorflow?

### ISSUE THREAD

**Issue by zuokai** [tensorflow/tensorflow#16192]
when i set the label=-1, there is an error: Received a label value of -1 which is outside the valid range of [0, 8)

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by zuokai** [comment]
Have I written custom code       yes
OS Platform and Distribution     centos6
TensorFlow installed from         github
TensorFlow version                   1.4
Bazel version    n/a
CUDA/cuDNN version 8.0
GPU model and memory n/a
Exact command to reproduce n/a

**Comment by tensorflowbutler** [comment]
The original poster has replied to this issue after the stat:awaiting response label was applied.

**Comment by zuokai** [comment]
# read images, labels
def read_img(path):
    imgs = []
    labels = []
    #label = []
 for line in open(path):
        label = []
        print(line)
        line = line.replace("\n","")
        strs = line.split('\t')
        imgs.append(strs[0])
        label.append(strs[1])
        label.append(strs[2])
                 if str(temp_str[1]) == "-1":
                    label.append("0")
                else:
                    label.append("1")
                if str(temp_str[2]) == "-1":
                    label.append("0")
                else:
                    label.append("1")
        labels.append(np.asarray(label, np.int32))
    return np.asarray(imgs, np.string_), np.asarray(labels, np.int32)
 
 
（3）split labels and ignore_weight
  y_ = tf.squeeze(tf.slice(y_all, [0, 0], [-1, 1]))
  y_gb_ = tf.squeeze(tf.slice(y_all, [0, 1], [-1, 1]))
    w_y = tf.squeeze(tf.slice(y_all, [0, 2], [-1, 1]))      #label1 weight
    w_y_gb = tf.squeeze(tf.slice(y_all, [0, 3], [-1, 1]))  #label2 weight
（4）use tf.losses.sparse_softmax_cross_entropy
        y_s_ = y_ * w_y
        y_gb_s_ = y_gb_ * w_y_gb
        y, y_gb = net.squeezenet(x, True, 0.999, CLASSES, CLASSES_FOOD)
        
        cross_entropy_gb = tf.losses.sparse_softmax_cross_entropy(labels=y_gb_s_, logits=y_gb, weights=w_y_gb)
         
        cross_entropy = tf.losses.sparse_softmax_cross_entropy(labels=y_s_, logits=y, weights = w_y)
       
        self.cost = tf.reduce_mean(cross_entropy, name='cross_entropy_loss') + tf.reduce_mean(cross_entropy_gb, name='cross_entropy_loss_gb')
